### ISSUE TITLE
[Q] init::LateResources{}; unsafe or other solution?

### ISSUE THREAD

**Issue by sbechet** [rtfm-rs/cortex-m-rtfm#308]
Hello,

I have a memory problem on stm32f103 hardware where memory is very small (20K).

As you know after compilation `main()` include `init()` then call `idle()`.

First time, we create local stack variable in `init()` then, at end, a `memcpy()` like copy stack data to rtfm reserved space `struct Resources`.

The local variables of `idle()` are therefore just after the local variables of `main()` which due to the low memory space is found right in the middle of the space reserved for the `struct Resource` :-1: 

What is the right solution?

A hack could be to not include `init()` in the `main()` to free the stack used for the initialization of `Resource` before calling `idle()` but is it probably an optimization of the compiler?

Another solution could be to directly initialize the data of the `Resource` structure, but in this case, I imagine that there is no good solution without unsafe?

In short, with you a good idea?

[source](/abelykh0/stm32f103-vga-rs/issues/1)

**Comment by sbechet** [comment]
Found a solution using Resource init and not LateResource.
